src/queue/session.rs

```diff
@@ -102,29 +113,51 @@ impl AuthQueue {
 
             Session::clear_cache(clear_cache_sessions, redis).await?;
 
-            let mut clear_cache_pats = Vec::new();
-
-            for id in pat_queue {
-                clear_cache_pats.push((Some(id), None, None));
-
-                sqlx::query!(
-                    "
-                    UPDATE pats
-                    SET last_used = $2
-                    WHERE (id = $1)
-                    ",
-                    id as PatId,
-                    Utc::now(),
-                )
-                .execute(&mut *transaction)
-                .await?;
-            }
+            let ids = pat_queue.iter().map(|id| id.0).collect_vec();
+            let clear_cache_pats = pat_queue
+                .into_iter()
+                .map(|id| (Some(id), None, None))
+                .collect_vec();
+            sqlx::query!(
+                "
+                UPDATE pats
+                SET last_used = $2
+                WHERE id IN
+                (SELECT * FROM UNNEST($1::bigint[]))
+                ",
+                &ids[..],
+                Utc::now(),
+            )
+            .execute(&mut *transaction)
+            .await?;
 
             PersonalAccessToken::clear_cache(clear_cache_pats, redis).await?;
 
+            process_oauth_access_tokens(oauth_access_token_queue, &mut transaction).await?;
+
             transaction.commit().await?;
         }
 
         Ok(())
     }
 }
+
+async fn process_oauth_access_tokens(
```

Comment:
why make this is a separate method? it doesn't seem to be used anywhere else

Reply:
I think the code in this file would be better if the `index` method called a small number of individual methods for the different things it does - `set_last_login_data`, `remove_expired_sessions`, `update_pat_last_used`, and `updated_oauth_access_token_last_used`. Then you can read the top level method and understand what's happening quickly, while being able to drill down and focus on one of the bits of functionality if you need to.

Pulling this chunk into a method was a step towards that, without refactoring the whole file.

Reply:
I'll be renaming this method to `updated_oauth_access_token_last_used`, which I think is better.

Reply:
Renamed